packages/next/build/webpack-config.ts

```diff
@@ -98,6 +98,14 @@ const BABEL_CONFIG_FILES = [
   'babel.config.cjs',
 ]
 
+function appDirIssuerLayer(layer: string) {
+  return (
+    layer === WEBPACK_LAYERS.client ||
+    layer === WEBPACK_LAYERS.server ||
+    layer === WEBPACK_LAYERS.appClient
```

Comment:
```suggestion
```

I think we can warn in the server compiler only, otherwise people will see it twice.

Reply:
we have to alias in appClient layer for the case that legacy head doesn't show up in client bundle, added test for it